From b70616d340c2914cf076a7b868578da903549aeb Mon Sep 17 00:00:00 2001
From: hugo SERGENT <hugo.sergent@st.com>
Date: Tue, 2 Apr 2019 09:26:41 +0200
Subject: [PATCH] Makefile: Update makefiles to match with the layer
 compilation

little update of the .jsons to delete the commenataries
in order to use a script based on jq
Signed-off-by: hugo SERGENT <hugo.sergent@st.com>
---
 Makefile                      |   2 +-
 lora_pkt_fwd/Makefile         |  18 +++---
 lora_pkt_fwd/global_conf.json | 128 ++++++++++++++++--------------------------
 lora_pkt_fwd/local_conf.json  |  14 +++--
 4 files changed, 67 insertions(+), 95 deletions(-)

diff --git a/Makefile b/Makefile
index aee59c7..8fa4a64 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 ### Environment constants 
 
-LGW_PATH ?= ../../lora_gateway/libloragw
+LGW_PATH ?= ../../recipe-sysroot/usr
 ARCH ?=
 CROSS_COMPILE ?=
 export
diff --git a/lora_pkt_fwd/Makefile b/lora_pkt_fwd/Makefile
index 1330d62..c61ca4c 100644
--- a/lora_pkt_fwd/Makefile
+++ b/lora_pkt_fwd/Makefile
@@ -4,7 +4,7 @@ APP_NAME := lora_pkt_fwd
 
 ### Environment constants 
 
-LGW_PATH ?= ../../lora_gateway/libloragw
+LGW_PATH ?= ../../recipe-sysroot/usr
 ARCH ?=
 CROSS_COMPILE ?=
 
@@ -14,7 +14,7 @@ INCLUDES = $(wildcard inc/*.h)
 ### External constant definitions
 # must get library build option to know if mpsse must be linked or not
 
-include $(LGW_PATH)/library.cfg
+include $(LGW_PATH)/include/library.cfg
 RELEASE_VERSION := `cat ../VERSION`
 
 ### Constant symbols
@@ -29,12 +29,12 @@ VFLAG := -D VERSION_STRING="\"$(RELEASE_VERSION)\""
 # List the library sub-modules that are used by the application
 
 LGW_INC =
-ifneq ($(wildcard $(LGW_PATH)/inc/config.h),)
+ifneq ($(wildcard $(LGW_PATH)/include/config.h),)
   # only for HAL version 1.3 and beyond
-  LGW_INC += $(LGW_PATH)/inc/config.h
+  LGW_INC += $(LGW_PATH)/include/config.h
 endif
-LGW_INC += $(LGW_PATH)/inc/loragw_hal.h
-LGW_INC += $(LGW_PATH)/inc/loragw_gps.h
+LGW_INC += $(LGW_PATH)/include/loragw_hal.h
+LGW_INC += $(LGW_PATH)/include/loragw_gps.h
 
 ### Linking options
 
@@ -54,14 +54,14 @@ $(OBJDIR):
 	mkdir -p $(OBJDIR)
 
 $(OBJDIR)/%.o: src/%.c $(INCLUDES) | $(OBJDIR)
-	$(CC) -c $(CFLAGS) -I$(LGW_PATH)/inc $< -o $@
+	$(CC) -c $(CFLAGS) -I$(LGW_PATH)/include $< -o $@
 
 ### Main program compilation and assembly
 
 $(OBJDIR)/$(APP_NAME).o: src/$(APP_NAME).c $(LGW_INC) $(INCLUDES) | $(OBJDIR)
-	$(CC) -c $(CFLAGS) $(VFLAG) -I$(LGW_PATH)/inc $< -o $@
+	$(CC) -c $(CFLAGS) $(VFLAG) -I$(LGW_PATH)/include $< -o $@
 
-$(APP_NAME): $(OBJDIR)/$(APP_NAME).o $(LGW_PATH)/libloragw.a $(OBJDIR)/parson.o $(OBJDIR)/base64.o $(OBJDIR)/jitqueue.o $(OBJDIR)/timersync.o
+$(APP_NAME): $(OBJDIR)/$(APP_NAME).o $(LGW_PATH)/lib/libloragw.a $(OBJDIR)/parson.o $(OBJDIR)/base64.o $(OBJDIR)/jitqueue.o $(OBJDIR)/timersync.o
 	$(CC) -L$(LGW_PATH) $< $(OBJDIR)/parson.o $(OBJDIR)/base64.o $(OBJDIR)/jitqueue.o $(OBJDIR)/timersync.o -o $@ $(LIBS)
 
 ### EOF
diff --git a/lora_pkt_fwd/global_conf.json b/lora_pkt_fwd/global_conf.json
index 2d0948c..a077317 100644
--- a/lora_pkt_fwd/global_conf.json
+++ b/lora_pkt_fwd/global_conf.json
@@ -1,26 +1,14 @@
 {
     "SX1301_conf": {
         "lorawan_public": true,
-        "clksrc": 1, /* radio_1 provides clock to concentrator */
-        "lbt_cfg": {
-            "enable": false,
-            "rssi_target": -80, /* dBm */
-            "chan_cfg":[ /* 8 channels maximum */
-                { "freq_hz": 867100000, "scan_time_us": 128 },
-                { "freq_hz": 867300000, "scan_time_us": 5000 },
-                { "freq_hz": 867500000, "scan_time_us": 128 },
-                { "freq_hz": 869525000, "scan_time_us": 128 }
-            ],
-            "sx127x_rssi_offset": -4 /* dB */
-        },
-        "antenna_gain": 0, /* antenna gain, in dBi */
+        "clksrc": 1,
+        "antenna_gain": 0,
         "radio_0": {
             "enable": true,
             "type": "SX1257",
             "freq": 867500000,
             "rssi_offset": -166.0,
             "tx_enable": true,
-            "tx_notch_freq": 129000, /* [126..250] KHz */
             "tx_freq_min": 863000000,
             "tx_freq_max": 870000000
         },
@@ -32,55 +20,46 @@
             "tx_enable": false
         },
         "chan_multiSF_0": {
-            /* Lora MAC channel, 125kHz, all SF, 868.1 MHz */
             "enable": true,
             "radio": 1,
             "if": -400000
         },
         "chan_multiSF_1": {
-            /* Lora MAC channel, 125kHz, all SF, 868.3 MHz */
             "enable": true,
             "radio": 1,
             "if": -200000
         },
         "chan_multiSF_2": {
-            /* Lora MAC channel, 125kHz, all SF, 868.5 MHz */
             "enable": true,
             "radio": 1,
             "if": 0
         },
         "chan_multiSF_3": {
-            /* Lora MAC channel, 125kHz, all SF, 867.1 MHz */
             "enable": true,
             "radio": 0,
             "if": -400000
         },
         "chan_multiSF_4": {
-            /* Lora MAC channel, 125kHz, all SF, 867.3 MHz */
             "enable": true,
             "radio": 0,
             "if": -200000
         },
         "chan_multiSF_5": {
-            /* Lora MAC channel, 125kHz, all SF, 867.5 MHz */
             "enable": true,
             "radio": 0,
             "if": 0
         },
         "chan_multiSF_6": {
-            /* Lora MAC channel, 125kHz, all SF, 867.7 MHz */
             "enable": true,
             "radio": 0,
             "if": 200000
         },
         "chan_multiSF_7": {
-            /* Lora MAC channel, 125kHz, all SF, 867.9 MHz */
             "enable": true,
             "radio": 0,
             "if": 400000
         },
         "chan_Lora_std": {
-            /* Lora MAC channel, 250kHz, SF7, 868.3 MHz */
             "enable": true,
             "radio": 1,
             "if": -200000,
@@ -88,7 +67,6 @@
             "spread_factor": 7
         },
         "chan_FSK": {
-            /* FSK 50kbps channel, 868.8 MHz */
             "enable": true,
             "radio": 1,
             "if": 300000,
@@ -96,133 +74,123 @@
             "datarate": 50000
         },
         "tx_lut_0": {
-            /* TX gain table, index 0 */
             "pa_gain": 0,
             "mix_gain": 8,
             "rf_power": -6,
             "dig_gain": 0
         },
         "tx_lut_1": {
-            /* TX gain table, index 1 */
             "pa_gain": 0,
             "mix_gain": 10,
             "rf_power": -3,
             "dig_gain": 0
         },
         "tx_lut_2": {
-            /* TX gain table, index 2 */
             "pa_gain": 0,
             "mix_gain": 12,
             "rf_power": 0,
             "dig_gain": 0
         },
         "tx_lut_3": {
-            /* TX gain table, index 3 */
             "pa_gain": 1,
             "mix_gain": 8,
             "rf_power": 3,
             "dig_gain": 0
         },
         "tx_lut_4": {
-            /* TX gain table, index 4 */
             "pa_gain": 1,
             "mix_gain": 10,
             "rf_power": 6,
             "dig_gain": 0
         },
         "tx_lut_5": {
-            /* TX gain table, index 5 */
             "pa_gain": 1,
             "mix_gain": 12,
             "rf_power": 10,
             "dig_gain": 0
         },
         "tx_lut_6": {
-            /* TX gain table, index 6 */
             "pa_gain": 1,
             "mix_gain": 13,
             "rf_power": 11,
             "dig_gain": 0
         },
         "tx_lut_7": {
-            /* TX gain table, index 7 */
             "pa_gain": 2,
             "mix_gain": 9,
             "rf_power": 12,
             "dig_gain": 0
         },
         "tx_lut_8": {
-            /* TX gain table, index 8 */
             "pa_gain": 1,
             "mix_gain": 15,
             "rf_power": 13,
             "dig_gain": 0
         },
         "tx_lut_9": {
-            /* TX gain table, index 9 */
             "pa_gain": 2,
             "mix_gain": 10,
             "rf_power": 14,
             "dig_gain": 0
         },
         "tx_lut_10": {
-            /* TX gain table, index 10 */
             "pa_gain": 2,
             "mix_gain": 11,
             "rf_power": 16,
             "dig_gain": 0
         },
         "tx_lut_11": {
-            /* TX gain table, index 11 */
             "pa_gain": 3,
             "mix_gain": 9,
             "rf_power": 20,
             "dig_gain": 0
-        },
-        "tx_lut_12": {
-            /* TX gain table, index 12 */
-            "pa_gain": 3,
-            "mix_gain": 10,
-            "rf_power": 23,
-            "dig_gain": 0
-        },
-        "tx_lut_13": {
-            /* TX gain table, index 13 */
-            "pa_gain": 3,
-            "mix_gain": 11,
-            "rf_power": 25,
-            "dig_gain": 0
-        },
-        "tx_lut_14": {
-            /* TX gain table, index 14 */
-            "pa_gain": 3,
-            "mix_gain": 12,
-            "rf_power": 26,
-            "dig_gain": 0
-        },
-        "tx_lut_15": {
-            /* TX gain table, index 15 */
-            "pa_gain": 3,
-            "mix_gain": 14,
-            "rf_power": 27,
-            "dig_gain": 0
+	},
+	"tx_lut_12": {
+		"desc": "TX gain table, index 12",
+		"pa_gain": 3,
+		"mix_gain": 10,
+		"rf_power": 23,
+		"dig_gain": 0
+	},
+	"tx_lut_13": {
+		"desc": "TX gain table, index 13",
+		"pa_gain": 3,
+		"mix_gain": 11,
+		"rf_power": 25,
+		"dig_gain": 0
+	},
+	"tx_lut_14": {
+		"desc": "TX gain table, index 14",
+		"pa_gain": 3,
+		"mix_gain": 12,
+		"rf_power": 26,
+		"dig_gain": 0
+	},
+	"tx_lut_15": {
+		"desc": "TX gain table, index 15",
+		"pa_gain": 3,
+		"mix_gain": 14,
+		"rf_power": 27,
+		"dig_gain": 0
         }
     },
-
-    "gateway_conf": {
-        "gateway_ID": "AA555A0000000000",
-        /* change with default server address/ports, or overwrite in local_conf.json */
-        "server_address": "localhost",
-        "serv_port_up": 1680,
-        "serv_port_down": 1680,
-        /* adjust the following parameters for your network */
-        "keepalive_interval": 10,
-        "stat_interval": 30,
-        "push_timeout_ms": 100,
-        /* forward only valid packets */
-        "forward_crc_valid": true,
-        "forward_crc_error": false,
-        "forward_crc_disabled": false
-    }
+	"gateway_conf": {
+		"gateway_ID": "AA555A0000000000",
+		"server_address": "localhost",
+		"serv_port_up": 1700,
+		"serv_port_down": 1700,
+		"keepalive_interval": 10,
+		"stat_interval": 30,
+		"push_timeout_ms": 100,
+		"forward_crc_valid": true,
+		"forward_crc_error": false,
+		"forward_crc_disabled": false,
+		"gps": false,
+		"gps_tty_path": "/dev/ttyAMA0",
+		"fake_gps": false,
+		"ref_latitude": 10,
+		"ref_longitude": 20,
+		"ref_altitude": 1
+	}
 }
 
diff --git a/lora_pkt_fwd/local_conf.json b/lora_pkt_fwd/local_conf.json
index cf6ff56..1dd63a4 100644
--- a/lora_pkt_fwd/local_conf.json
+++ b/lora_pkt_fwd/local_conf.json
@@ -1,7 +1,11 @@
 {
-/* Put there parameters that are different for each gateway (eg. pointing one gateway to a test server while the others stay in production) */
-/* Settings defined in global_conf will be overwritten by those in local_conf */
-    "gateway_conf": {
-        "gateway_ID": "AA555A0000000101" /* you must pick a unique 64b number for each gateway (represented by an hex string) */
-    }
+	"gateway_conf": {
+		"gateway_ID": "AA555A0000000000",
+		"serv_port_up": 1700,
+		"serv_port_down": 1700,
+		"serv_enabled": true,
+		"ref_latitude": 0,
+		"ref_longitude": 0,
+		"ref_altitude": 0 
+	}
 }
-- 
2.7.4

